diff --git a/Silicon/Ampere/AmpereAltraPkg/Drivers/RootComplexConfigDxe/RootComplexConfigDxe.c b/Silicon/Ampere/AmpereAltraPkg/Drivers/RootComplexConfigDxe/RootComplexConfigDxe.c
index e03be2a2f9..44b926cbdb 100644
--- a/Silicon/Ampere/AmpereAltraPkg/Drivers/RootComplexConfigDxe/RootComplexConfigDxe.c
+++ b/Silicon/Ampere/AmpereAltraPkg/Drivers/RootComplexConfigDxe/RootComplexConfigDxe.c
@@ -723,10 +723,6 @@ PcieRCScreenSetup (
     //
     OptionsOpCodeHandle = CreateDevMapOptions (RootComplex);
 
-    if (RootComplex->DefaultDevMapLow != 0) {
-      QuestionFlags |= EFI_IFR_FLAG_READ_ONLY;
-    }
-
     HiiCreateOneOfOpCode (
       StartOpCodeHandle,                        // Container for dynamic created opcodes
       0x8003 + MAX_EDITABLE_ELEMENTS * RCIndex, // Question ID (or call it "key")
