diff --git a/Makefile b/Makefile
index c9970e5..9262e70 100755
--- a/Makefile
+++ b/Makefile
@@ -8,6 +8,10 @@
 #
 SHELL := /bin/bash
 
+define calc
+$(shell echo $$(($(1))))
+endef
+
 # Default Input variables
 ATF_TBB ?= 1
 BUILD_LINUXBOOT ?= 0
@@ -72,7 +76,7 @@ DEFAULT_IASL_VER := $(shell $(PARSE_PLATFORMS_TOOL) -c $(PLATFORMS_CONFIG) -p $(
 IASL_VER ?= $(if $(VER_GT_104),$(DEFAULT_IASL_VER),20200110)
 
 # efitools version
-EFITOOLS_VER := 1.8.1
+EFITOOLS_VER := 1.9.2
 
 # File path variables
 LINUXBOOT_FMT := $(if $(shell echo $(BUILD_LINUXBOOT) | grep -w 1),_linuxboot,)
@@ -210,7 +214,9 @@ _check_efitools:
 	$(eval EFITOOLS_REPO_URL := https://github.com/vathpela/efitools.git)
 	$(eval export PATH := $(EFI_TOOLS_DIR):$(PATH))
 
-	@if which $(CERT_TO_EFI_SIG_LIST) &>/dev/null && which $(SIGN_EFI_SIG_LIST) &>/dev/null && $(CERT_TO_EFI_SIG_LIST) --version 2>/dev/null | grep $(EFITOOLS_VER); then \
+	@if which $(CERT_TO_EFI_SIG_LIST) &>/dev/null && which $(SIGN_EFI_SIG_LIST) &>/dev/null && \
+		[ `$(CERT_TO_EFI_SIG_LIST) --version | cut -d' ' -f2 | tr -d .` -ge `echo $(EFITOOLS_VER) | tr -d .` ]; \
+	then \
 		echo "OK"; \
 	else \
 		echo -e "Not Found\nDownloading and building efitools..."; \
@@ -249,7 +255,7 @@ _check_board_setting:
 _tianocore_prepare: _check_source _check_tools _check_compiler _check_iasl
 	$(if $(wildcard $(EDK2_SRC_DIR)/BaseTools/Source/C/bin),,$(MAKE) -C $(EDK2_SRC_DIR)/BaseTools -j $(NUM_THREADS))
 	$(eval export WORKSPACE := $(CUR_DIR))
-	$(eval export PACKAGES_PATH := $(shell echo $(REQUIRE_EDK2_SRC) | sed 's/ /:/g'))
+	$(eval export PACKAGES_PATH ?= $(shell echo $(REQUIRE_EDK2_SRC) | sed 's/ /:/g'))
 	$(eval export $(EDK2_GCC_TAG)_AARCH64_PREFIX := $(CROSS_COMPILE))
 	$(eval EDK2_FV_DIR := $(WORKSPACE)/Build/$(BOARD_NAME_UFL)/$(BUILD_VARIANT)_$(EDK2_GCC_TAG)/FV)
 
@@ -312,13 +318,13 @@ tianocore_img: _check_atf_tools _check_atf_slim _check_board_setting tianocore_f
 	@echo "Build Tianocore $(BUILD_VARIANT_UFL) Image - ATF VERSION: $(ATF_MAJOR).$(ATF_MINOR).$(ATF_BUILD)..."
 	$(eval DBB_KEY := $(EDK2_PLATFORMS_SRC_DIR)/Platform/Ampere/$(BOARD_NAME_UFL)Pkg/TestKeys/Dbb_AmpereTest.priv.pem)
 	@dd bs=1024 count=2048 if=/dev/zero | tr "\000" "\377" > $(OUTPUT_RAW_IMAGE)
-	@dd bs=1 seek=0 conv=notrunc if=$(ATF_SLIM) of=$(OUTPUT_RAW_IMAGE)
+	@dd bs=1024 seek=0 conv=notrunc if=$(ATF_SLIM) of=$(OUTPUT_RAW_IMAGE)
 	@if [ $(MAJOR_VER)$(MINOR_VER) -gt 202 ]; then \
 		$(CERTTOOL) -n --ntfw-nvctr 0 --key-alg rsa --hash-alg sha384 --nt-fw-key $(DBB_KEY) --nt-fw-cert ${ATF_SLIM}.crt --nt-fw ${ATF_SLIM}; \
-		dd bs=1 seek=1572864 conv=notrunc if=${ATF_SLIM}.crt of=${OUTPUT_RAW_IMAGE}; \
+		dd bs=$(call calc,0x10000) seek=$(call calc,0x180000/0x10000) conv=notrunc if=${ATF_SLIM}.crt of=${OUTPUT_RAW_IMAGE}; \
 		rm -f ${ATF_SLIM}.crt; \
 	fi
-	@dd bs=1 seek=2031616 conv=notrunc if=$(OUTPUT_BOARD_SETTING_BIN) of=$(OUTPUT_RAW_IMAGE)
+	@dd bs=$(call calc,0x10000) seek=$(call calc,0x1F0000/0x10000) conv=notrunc if=$(OUTPUT_BOARD_SETTING_BIN) of=$(OUTPUT_RAW_IMAGE)
 
 	@if [ $(ATF_TBB) -eq 1 ]; then \
 		$(MAKE) -C $(SCRIPTS_DIR) _tianocore_sign_fd; \
@@ -332,7 +338,7 @@ tianocore_img: _check_atf_tools _check_atf_slim _check_board_setting tianocore_f
 # compatibility with the support of firmware update utility.
 	@if [ $(ATF_MAJOR)$(ATF_MINOR) -eq 103 ] || [ $(ATF_MAJOR)$(ATF_MINOR) -eq 201 ]; then \
 		dd if=/dev/zero bs=1024 count=4096 | tr "\000" "\377" > $(OUTPUT_IMAGE); \
-		dd bs=1 seek=4194304 conv=notrunc if=$(OUTPUT_RAW_IMAGE) of=$(OUTPUT_IMAGE); \
+		dd bs=$(call calc,0x10000) seek=$(call calc,0x400000/0x10000) conv=notrunc if=$(OUTPUT_RAW_IMAGE) of=$(OUTPUT_IMAGE); \
 	else \
 		cp $(OUTPUT_RAW_IMAGE) $(OUTPUT_IMAGE); \
 	fi
@@ -357,8 +363,8 @@ tianocore_capsule: tianocore_img dbukeys_auth
 			ln -sf $(realpath $(SCP_SLIM)) $(SCP_IMAGE); \
 		else \
 			echo "Append dummy data to origin SCP image"; \
-			dd bs=1 count=261632 if=/dev/zero | tr "\000" "\377" > $(SCP_IMAGE).append; \
-			dd bs=1 seek=0 conv=notrunc if=$(SCP_SLIM) of=$(SCP_IMAGE).append; \
+			dd bs=$(call calc,0x100) count=$(call calc,0x3FE00/0x100) if=/dev/zero | tr "\000" "\377" > $(SCP_IMAGE).append; \
+			dd bs=$(call calc,0x100) seek=0 conv=notrunc if=$(SCP_SLIM) of=$(SCP_IMAGE).append; \
 			openssl dgst -sha384 -sign $(DBU_KEY) -out $(SCP_IMAGE).sig $(SCP_IMAGE).append; \
 			cat $(SCP_IMAGE).sig $(SCP_IMAGE).append > $(SCP_IMAGE).signed; \
 			cp -r $(SCP_IMAGE).signed $(SCP_IMAGE); \
@@ -383,8 +389,8 @@ tianocore_capsule: tianocore_img dbukeys_auth
 	else \
 		echo "Sign Tianocore Image"; \
 		echo "Append to dummy byte to UEFI image"; \
-		dd bs=1 count=13630976 if=/dev/zero | tr "\000" "\377" > $(OUTPUT_RAW_IMAGE).append; \
-		dd bs=1 seek=0 conv=notrunc if=$(OUTPUT_RAW_IMAGE) of=$(OUTPUT_RAW_IMAGE).append; \
+		dd bs=$(call calc,0x100) count=$(call calc,0xCFFE00/0x100) if=/dev/zero | tr "\000" "\377" > $(OUTPUT_RAW_IMAGE).append; \
+		dd bs=$(call calc,0x100) seek=0 conv=notrunc if=$(OUTPUT_RAW_IMAGE) of=$(OUTPUT_RAW_IMAGE).append; \
 		openssl dgst -sha384 -sign $(DBU_KEY) -out $(OUTPUT_RAW_IMAGE).sig $(OUTPUT_RAW_IMAGE).append; \
 		cat $(OUTPUT_RAW_IMAGE).sig $(OUTPUT_RAW_IMAGE).append > $(OUTPUT_RAW_IMAGE).signed; \
 		ln -sf $(OUTPUT_RAW_IMAGE).signed $(TIANOCORE_ATF_IMAGE); \
